nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/admin/SetHealthCheckServiceConfiguration.java

```diff
@@ -0,0 +1,402 @@
+        }
+    }
+
+    private void configureDynamically(BaseThresholdHealthCheck<?, ?> service) {
+        if (thresholdCritical != null) {
+            service.getOptions().setThresholdCritical(Integer.valueOf(thresholdCritical));
+            report.appendMessage(strings.getLocalString(
+                    "healthcheck.service.configure.threshold.critical.success",
+                    "Critical threshold for {0} service is set with value {1}.", serviceName, thresholdCritical));
+            report.appendMessage("\n");
+        }
+        if (thresholdWarning != null) {
+            service.getOptions().setThresholdWarning(Integer.valueOf(thresholdWarning));
+            report.appendMessage(strings.getLocalString("healthcheck.service.configure.threshold.warning.success",
+                    "Warning threshold for {0} service is set with value {1}.", serviceName, thresholdWarning));
+            report.appendMessage("\n");
+        }
+        if (thresholdGood != null) {
+            service.getOptions().setThresholdGood(Integer.valueOf(thresholdGood));
+            report.appendMessage(strings.getLocalString("healthcheck.service.configure.threshold.good.success",
+                    "Good threshold for {0} service is set with value {1}.", serviceName, thresholdGood));
+            report.appendMessage("\n");
+        }
+    }
+
+    private <C extends Checker> Checker updateProperties(Checker config, Class<C> type) throws PropertyVetoException {
+        updateProperty(config, "enabled", config.getEnabled(), String.valueOf(enabled), Checker::setEnabled);
+        updateProperty(config, "time", config.getTime(), time, Checker::setTime);
+        updateProperty(config, "time-unit", config.getUnit(), unit, Checker::setUnit);
+        if (HoggingThreadsChecker.class.isAssignableFrom(type)) {
+            HoggingThreadsChecker htConfig = (HoggingThreadsChecker) config;
+            updateProperty(htConfig, "hogging-threads-threshold", htConfig.getThresholdPercentage(), 
+                    hogginThreadsThreshold, HoggingThreadsChecker::setThresholdPercentage);
+            updateProperty(htConfig, "hogging-threads-retry-count", htConfig.getRetryCount(), 
+                    hogginThreadsRetryCount, HoggingThreadsChecker::setRetryCount);
+        }
+        if (StuckThreadsChecker.class.isAssignableFrom(type)) {
+            StuckThreadsChecker stConfig = (StuckThreadsChecker) config;
+            updateProperty(stConfig, "stuck-threads-threshold", stConfig.getThreshold(), 
+                    stuckThreadsThreshold, StuckThreadsChecker::setThreshold);
+            updateProperty(stConfig, "stuck-threads-threshold-unit", stConfig.getThresholdTimeUnit(), 
+                    stuckThreadsThresholdUnit, StuckThreadsChecker::setThresholdTimeUnit);
+        }
+        return config;
+    }
+
+    private void updateProperty(ThresholdDiagnosticsChecker config, final String name, final String value)
```

Comment:
Again, a small bit of Javadoc explaining that this updateProperty method is specifically for one purpose.

Reply:
Added.